addons/hr_maintenance/wizard/hr_departure_wizard.py

```diff
@@ -0,0 +1,14 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class HrDepartureWizard(models.TransientModel):
+    _inherit = 'hr.departure.wizard'
+
+    unassign_equipment = fields.Boolean("Free Equiments", default=True, help="Unassign Employee from Equipments")
+
+    def action_register_departure(self):
+        super().action_register_departure()
+        if self.unassign_equipment:
+            self.employee_id.equipment_ids.write({'employee_id': False})
```

Comment:
and for this - https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html?highlight=orm#odoo.fields.Command

For one2many and many2many we need to use special command